eclair-core/src/main/scala/fr/acinq/eclair/api/Service.scala

```diff
@@ -1,422 +1,222 @@
-                      }
-
-                        // check received payments
-                        case "checkpayment" => req.params match {
-                          case JString(identifier) :: Nil => completeRpcFuture(req.id, for {
-                            paymentHash <- Try(PaymentRequest.read(identifier)) match {
-                              case Success(pr) => Future.successful(pr.paymentHash)
-                              case _ => Try(ByteVector.fromValidHex(identifier)) match {
-                                case Success(s) => Future.successful(s)
-                                case _ => Future.failed(new IllegalArgumentException("payment identifier must be a payment request or a payment hash"))
-                              }
-                            }
-                            found <- (paymentHandler ? CheckPayment(ByteVector32.fromValidHex(identifier))).map(found => new JBool(found.asInstanceOf[Boolean]))
-                          } yield found)
-                          case _ => reject(UnknownParamsRejection(req.id, "[paymentHash] or [paymentRequest]"))
-                        }
-
-                        // retrieve audit events
-                        case "audit" =>
-                          val (from, to) = req.params match {
-                            case JInt(from) :: JInt(to) :: Nil => (from.toLong, to.toLong)
-                            case _ => (0L, Long.MaxValue)
-                          }
-                          completeRpcFuture(req.id, Future(AuditResponse(
-                            sent = nodeParams.auditDb.listSent(from, to),
-                            received = nodeParams.auditDb.listReceived(from, to),
-                            relayed = nodeParams.auditDb.listRelayed(from, to))
-                          ))
-
-                        case "networkfees" =>
-                          val (from, to) = req.params match {
-                            case JInt(from) :: JInt(to) :: Nil => (from.toLong, to.toLong)
-                            case _ => (0L, Long.MaxValue)
-                          }
-                          completeRpcFuture(req.id, Future(nodeParams.auditDb.listNetworkFees(from, to)))
-
-                        // retrieve fee stats
-                        case "channelstats" => completeRpcFuture(req.id, Future(nodeParams.auditDb.stats))
-
-
-                        // method name was not found
-                        case _ => reject(UnknownMethodRejection(req.id))
-                      }
+              post {
+                path("getinfo") {
+                  complete(eclairApi.getInfoResponse())
+                } ~
+                  path("connect") {
+                    formFields("uri".as[String]) { uri =>
+                      complete(eclairApi.connect(uri))
+                    } ~ formFields("nodeId".as[PublicKey], "host".as[String], "port".as[Int]) { (nodeId, host, port) =>
```

Comment:
Could the port be optional, with default value being `9735`?

We can retrieve the default port value from `fr.acinq.eclair.io.NodeURI#DEFAULT_PORT`.

Reply:
Done in 23c3515

Reply:
Also updated the doc: https://acinq.github.io/eclair/#connect-manually